### PR TITLE
feat: adding brakeman.ignore support to huskyci

### PR DESCRIPTION
**Pull request by GabhenDM**
### Description

This PR aims to add brakeman.ignore file support to HuskyCI.

Since there is no current implementation for utilizing "#nohusky" as referenced in #508  adding support to brakeman.ignore will help dealing with false-positives on the CI flow.

### Proposed Changes

Adding to the api/config.yaml in the brakeman section a if statement verifying if there is a brakeman.ignore file on the target project root folder, if yes run brakeman with the file as a parameter.

It requires a bit of manual work to generate the ignore file, since it is only generated by brakeman itself, one would have to run it manually, validate the findings and ignore them as the tools [documentation](https://brakemanscanner.org/docs/ignoring_false_positives/)

### Testing
The branch of poc-ruby-brakeman on my [forked repo](https://github.com/GabhenDM/huskyCI/tree/poc-ruby-brakeman) contains a brakeman.ignore removing two other warnings that would show up at a HuskyCI analysis result.

Running this current build agaisnt that repo should be enough to validate the implementation. 

🐼

### CODE REVIEW

File: api/config.yaml [hunk 1]
```diff
@@ -129,8 +129,13 @@ brakeman:
     GIT_TERMINAL_PROMPT=0 git clone -b %GIT_BRANCH% --single-branch %GIT_REPO% code --quiet 2> /tmp/errorGitCloneBrakeman
     if [ $? -eq 0 ]; then
       if [ -d /code/app ]; then
-        brakeman -q -o results.json /code
-        jq -j -M -c . results.json
+        if [ -f /code/brakeman.ignore ]; then
```
Comment (tcarreira) [inline api/config.yaml:132]: Could this be rewritten as:
```
      if [ -d /code/app ]; then
        if [ -f /code/brakeman.ignore ]; then
          brakeman -q -i /code/brakeman.ignore -o results.json /code
        else
          brakeman -q -o results.json /code
        fi
          jq -j -M -c . results.json
      else
``` 

(less duplicated code)

**Review by tcarreira — CHANGES_REQUESTED** [review]

**Review by tcarreira — APPROVED** [review]
LGTM

**Review by gustavocovas — APPROVED** [review]

**Comment by gustavocovas** [comment]
Looks good to me! Do you think we need to have some integration tests, @joserenatosilva, @rafaveira3 or @Krlier?

**Comment by rafaveira3** [comment]
Hello, everyone! Glad to know that we are now able to skip a few tests in Brakeman! :smile: 

@gustavocovas, a good strategy here might be adding vulnerable code into [this branch](https://github.com/globocom/huskyCI/tree/poc-ruby-brakeman) and adding a `brakeman.ignore` as well. huskyCI should now skip them and not return a 190 status code.

Another topic we must be aware of is reporting this metric in the output as the "NoSecHusky" type. It would be a good idea saving it just like we do on other languages like Python, as follows:

```
[HUSKYCI][SUMMARY] Python -> huskyci/bandit:1.6.2
[HUSKYCI][SUMMARY] High: 0
[HUSKYCI][SUMMARY] Medium: 1
[HUSKYCI][SUMMARY] Low: 1
[HUSKYCI][SUMMARY] NoSecHusky: 1

[HUSKYCI][SUMMARY] Total
[HUSKYCI][SUMMARY] High: 0
[HUSKYCI][SUMMARY] Medium: 1
[HUSKYCI][SUMMARY] Low: 1
[HUSKYCI][SUMMARY] NoSecHusky: 1
```

**Comment by GabhenDM** [comment]
> Hello, everyone! Glad to know that we are now able to skip a few tests in Brakeman! smile
> 
> @gustavocovas, a good strategy here might be adding vulnerable code into [this branch](https://github.com/globocom/huskyCI/tree/poc-ruby-brakeman) and adding a `brakeman.ignore` as well. huskyCI should now skip them and not return a 190 status code.
> 
> Another topic we must be aware of is reporting this metric in the output as the "NoSecHusky" type. It would be a good idea saving it just like we do on other languages like Python, as follows:
> 
> ```
> [HUSKYCI][SUMMARY] Python -> huskyci/bandit:1.6.2
> [HUSKYCI][SUMMARY] High: 0
> [HUSKYCI][SUMMARY] Medium: 1
> [HUSKYCI][SUMMARY] Low: 1
> [HUSKYCI][SUMMARY] NoSecHusky: 1
> 
> [HUSKYCI][SUMMARY] Total
> [HUSKYCI][SUMMARY] High: 0
> [HUSKYCI][SUMMARY] Medium: 1
> [HUSKYCI][SUMMARY] Low: 1
> [HUSKYCI][SUMMARY] NoSecHusky: 1
> ```
Great idea! I checked that brakeman does output ignored warnings to it's json file, being quite easy to add as a nosechusky metric.

Will also open a PR to the poc-ruby-brakeman branch in order to add a brakeman.ignore test.

**Pull request merged**